tests/tests/test-precompile-assets-erc20.ts

```diff
@@ -0,0 +1,410 @@
+
+      assetId = context.polkadotApi.createType(
+        "AssetId",
+        new BN("42259045809535163221576417993425387648")
+      );
+      const assetDetails = context.polkadotApi.createType("AssetDetails", { supply: balance });
+
+      await mockAssetBalance(context, assetBalance, assetDetails, sudoAccount, assetId);
+
+      const contractData = await getCompiled("ERC20Instance");
+      iFace = new ethers.utils.Interface(contractData.contract.abi);
+      const { contract, rawTx } = await createContract(context.web3, "ERC20Instance");
+      const address = contract.options.address;
+      await context.createBlock({ transactions: [rawTx] });
+    });
+    it("allows to approve transfer and use transferFrom", async function () {
+      // Create approval
+      let data = iFace.encodeFunctionData(
+        // action
+        "approve",
+        [BALTATHAR, 1000]
+      );
+
+      let tx = await createTransaction(context.web3, {
+        from: ALITH,
+        privateKey: ALITH_PRIV_KEY,
+        value: "0x0",
+        gas: "0x200000",
+        gasPrice: GAS_PRICE,
+        to: ADDRESS_ERC20,
+        data: data,
+      });
+
+      let block = await context.createBlock({
+        transactions: [tx],
+      });
+
+      let approvals = (await context.polkadotApi.query.assets.approvals(
+        assetId,
+        ALITH,
+        BALTATHAR
+      )) as any;
+
+      expect(approvals.unwrap().amount.eq(new BN(1000))).to.equal(true);
+      // We are gonna spend 1000 from alith to send it to charleth
+      data = iFace.encodeFunctionData(
+        // action
+        "transferFrom",
+        [ALITH, CHARLETH, 1000]
+      );
+
+      tx = await createTransaction(context.web3, {
+        from: BALTATHAR,
+        privateKey: BALTATHAR_PRIV_KEY,
+        value: "0x0",
+        gas: "0x200000",
+        gasPrice: GAS_PRICE,
+        to: ADDRESS_ERC20,
+        data: data,
+      });
+
+      block = await context.createBlock({
+        transactions: [tx],
+      });
+      const receipt = await context.web3.eth.getTransactionReceipt(block.txResults[0].result);
+
+      expect(receipt.logs.length).to.eq(1);
+      expect(receipt.logs[0].address).to.eq(ADDRESS_ERC20);
+      expect(receipt.logs[0].topics.length).to.eq(3);
+      expect(receipt.logs[0].topics[0]).to.eq(SELECTORS.logTransfer);
+      expect(receipt.status).to.equal(true);
+
+      // Approve amount is null now
+      approvals = (await context.polkadotApi.query.assets.approvals(
+        assetId,
+        ALITH,
+        BALTATHAR
+      )) as any;
+      expect(approvals.isNone).to.eq(true);
+
+      // Charleth balance is 1000
+      let charletBalance = (await context.polkadotApi.query.assets.account(
+        assetId,
+        CHARLETH
+      )) as any;
+      expect(charletBalance.balance.eq(new BN(1000))).to.equal(true);
+    });
+  },
+  true
+);
+
+describeDevMoonbeam(
+  "Precompiles - Assets-ERC20 Wasm",
+  (context) => {
```

Comment:
You can execute tests in wasm by adding true in third parameter, see describeDevMoonbeam definition